src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportService.java

```diff
@@ -0,0 +1,146 @@
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.SupplierStatementRepository;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.model.EnvelopeSupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.SupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReport;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.SummaryReportItem;
+import uk.gov.hmcts.reform.blobrouter.services.EnvelopeService;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.groupingBy;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class SummaryReportService {
+
+    private static final Logger logger = getLogger(SummaryReportService.class);
+
+    private final SupplierStatementRepository repository;
+    private final ReconciliationReportRepository reconciliationReportRepository;
+    private final ObjectMapper objectMapper;
+    private final EnvelopeService envelopeService;
+    private final Map<String, StorageConfigItem> storageConfig; // container-specific configuration, by container name
+
+    public SummaryReportService(
+        SupplierStatementRepository repository,
+        ReconciliationReportRepository reconciliationReportRepository,
+        ObjectMapper objectMapper,
+        EnvelopeService envelopeService,
+        ServiceConfiguration serviceConfiguration
+    ) {
+        this.repository = repository;
+        this.reconciliationReportRepository = reconciliationReportRepository;
+        this.objectMapper = objectMapper;
+        this.envelopeService = envelopeService;
+        this.storageConfig =  serviceConfiguration.getStorageConfig();
+    }
+
+    public void process(LocalDate date) throws JsonProcessingException {
+        Optional<EnvelopeSupplierStatement> optSupplierStatement = repository.findLatest(date);
+
+        if (!optSupplierStatement.isPresent()) {
+            logger.error("No supplier statement found for {}", date);
+            return;
+        }
+
+        EnvelopeSupplierStatement envelopeSupplierStatement = optSupplierStatement.get();
+        SupplierStatement supplierStatement = objectMapper
+            .readValue(envelopeSupplierStatement.content, SupplierStatement.class);
+
+        var envelopeList = envelopeService.getEnvelopes(date);
+
+        if (CollectionUtils.isEmpty(envelopeList)) {
+            logger.info("No envelopes find for {} ", date);
+        }
+
+        Map<TargetStorageAccount, List<uk.gov.hmcts.reform.blobrouter.reconciliation.model.in.Envelope>>
+            supplierEnvelopesMap = supplierStatement
+            .envelopes
+            .stream()
+            .collect(groupingBy(e -> storageConfig.get(e.container).getTargetStorageAccount()));
```

Comment:
after grouping, you could map the values to instances of `SummaryReportItem` to avoid those pesky namespaces and later on you'll be able to compare two sets of items of the same models easily.

Reply:
SummaryReportItem for discrepancies.  I am planning to rename reconciliation `envelope` to sth else.